meta-iotqa/lib/oeqa/runtime/sanity/mraa_gpio.py

```diff
@@ -0,0 +1,44 @@
+from oeqa.oetest import oeRuntimeTest
+import unittest
+import subprocess
+from time import sleep
+
+class MraaGpioTest(oeRuntimeTest):
+    '''
+    These tests require to use BeagleBone as testing host
+    '''
+    pin = ""
+    def setUp(self):
+        (status, output)= self.target.run("mraa-gpio version")
+        output = output.lower()
+        if any(x in output for x in ("broxton", "tuchuck", "joule")):
+            self.pin = "51"
+        elif "minnowboard" in output:
+            self.pin = "25"
+        else:
+            raise unittest.SkipTest(output)
+
+    def test_gpio(self):
+        '''
+        Test a GPIO pin on and off and check the pin output with
+        BeagleBone
+        '''
+        def check_gpio_output():
+            cmd = "cat /sys/class/gpio/gpio20/value".split()
```

Comment:
No the GPIO pin has to be exported, but it's already exported on the Beaglebones. I added few lines to a boot script on the Beaglebone so it's always exported and ready after booting.